SopraGameAction.cpp

```diff
@@ -0,0 +1,197 @@
+                    //@TODO
+                } else {
+                    throw std::runtime_error("No player at specified interception point");
+                }
+            }
+        }
+    }
+
+    auto Shot::successProb(const gameModel::Environment &envi) const -> double{
+
+        if (gameModel::Environment::getCell(this->target) == gameModel::Cell::OutOfBounds){
+            return 0;
+        }
+        else {
+            // @ToDo: Wahrscheinlichkeit für das Abfangen einfügen.
+            return pow(envi.config.gameDynamicsProbs.throwSuccess,
+                       gameController::getDistance(this->actor.get()->position, this->target));
+        }
+    }
+
+    // fertig ?
+    auto Shot::check(const gameModel::Environment &envi) const -> ActionResult {
+        if (gameModel::Environment::getCell(this->target) == gameModel::Cell::OutOfBounds) {
+            return ActionResult::Impossible;
+        } else if(typeid(*actor) == typeid(gameModel::Seeker)){
+            return ActionResult::Impossible;
+        } else if(typeid(*actor) == typeid(gameModel::Keeper) || typeid(*actor) == typeid(gameModel::Chaser)){
+            if(envi.quaffle.position != actor->position){
+                return ActionResult::Impossible;
+            }
+        } else if(typeid(*actor) == typeid(gameModel::Beater)){
+            if(envi.bludgers[0].position != actor->position &&
+            envi.bludgers[1].position != actor->position){
+                return ActionResult::Impossible;
+            }
+        }
+
+        return ActionResult::Success;
+    }
+
+    auto Shot::executeAll(const gameModel::Environment &envi) const ->
+    std::vector<std::pair<gameModel::Environment, double>> {
+
+        std::vector<std::pair<gameModel::Environment, double>> resultVect;
+
+        // @toDo: da fehlt noch alles
+
+        return resultVect;
+    }
+
+    auto Shot::getInterceptionPositions(const gameModel::Environment &env) const -> std::vector<gameModel::Position>{
+        auto crossedCells = gameController::getAllCrossedCells(this->actor->position, target);
+        std::vector<gameModel::Position> ret;
+        for(const auto &cell : crossedCells){
+            for(const auto &player : env.getOpponents(*actor)){
+                if(player->position == cell){
+                    ret.emplace_back(cell);
+                }
+            }
+        }
+
+        return ret;
+    }
+
+    auto Shot::getAllLandingCells(const gameModel::Environment &env) const -> std::vector<gameModel::Position> {
+#warning Was tun bei n gerade? Dann lässt sich das Quadrat nicht mittig um target platzieren
+        int n = static_cast<int>(std::ceil(gameController::getDistance(actor->position, target) / 7));
+        std::vector<gameModel::Position> ret;
+        using Env = gameModel::Environment;
+        using Cell = gameModel::Cell;
+        auto players = env.getAllPlayers();
+        ret.reserve(n * n);
+        for(int x = target.x - n / 2; x < target.x + n / 2; x++){
```

Comment:
```suggestion
        for(int x = target.x - n / 2; x <= target.x + n / 2; x++){
```
Mit < wird die rechteste Spalte nicht abgedeckt